src/foundry/foundry.js/roll.d.ts

```diff
@@ -1,501 +1,505 @@
-   * Step 5: Classify all remaining strings
-   *
-   * @param formula - The original string expression to parse
-   * @param data - A data object used to substitute for attributes in the formula
-   * @returns A parsed array of RollTerm instances
-   */
-  static parse(formula: string, data: object): RollTerm[];
+    /**
+     * Alias for evaluate.
+     * @see Roll#evaluate
+     */
+    roll({ minimize, maximize, async }?: Partial<Options & { async: false }>): this;
+    roll({ minimize, maximize, async }?: Partial<Options & { async: true }>): Promise<this>;
+    roll({ minimize, maximize, async }?: Partial<Options>): this | Promise<this>;
 
-  /**
-   * Replace referenced data attributes in the roll formula with values from the provided data.
-   * Data references in the formula use the \@attr syntax and would reference the corresponding attr key.
-   *
-   * @param formula - The original formula within which to replace
-   * @param data    - The data object which provides replacements
-   * @param missing - The value that should be assigned to any unmatched keys.
-   *                  If null, the unmatched key is left as-is.
-   * @param warn    - Display a warning notification when encountering an un-matched key.
-   *                  (default: `false`)
-   */
-  static replaceFormulaData<D>(
-    formula: string,
-    data: D,
-    {
-      missing,
-      warn
-    }?: {
-      missing?: string;
-      warn?: boolean;
-    }
-  ): string;
+    /**
+     * Create a new Roll object using the original provided formula and data.
+     * Each roll is immutable, so this method returns a new Roll instance using the same data.
+     * @param options - Evaluation options passed to Roll#evaluate
+     * @returns A new Roll object, rolled using the same formula and data
+     */
+    reroll(options?: Partial<Options & { async: false }>): this;
+    reroll(options?: Partial<Options & { async: true }>): Promise<this>;
+    reroll(options?: Partial<Options>): this | Promise<this>;
 
-  /**
-   * Split a formula by identifying its outer-most parenthetical and math terms
-   * @param _formula - The raw formula to split
-   * @returns An array of terms, split on parenthetical terms
-   */
-  protected static _splitParentheses(_formula: string): string[];
+    /**
+     * A factory method which constructs a Roll instance using the default configured Roll class.
+     * @typeParam D - the type of data object against which to parse attributes within the formula
+     * @param formula - The formula used to create the Roll instance
+     * @param data    - The data object which provides component data for the formula
+     * @param options - Additional options which modify or describe this Roll
+     * @returns The constructed Roll instance
+     */
+    static create<D extends Record<string, unknown> = {}>(
+      formula: string,
+      data?: D,
+      options?: Partial<Options>
+    ): Roll<D>;
```

Comment:
This should pull the used `Roll` class from the `CONFIG`. If you think it's too complicated to implement that now, please create a TODO here.